src/scandeval/generation.py

```diff
@@ -321,6 +333,89 @@ def generate_single_iteration(
     return itr_scores
 
 
+def get_prefix_allowed_fn(
+    dataset_config: DatasetConfig, tokenizer: Tokenizer
+) -> Callable[[int, torch.Tensor], torch.Tensor] | None:
+    """Return the prefix allowed function to use for structured generation.
+
+    Args:
+        dataset_config:
+            The dataset config.
+        tokenizer:
+            The tokenizer.
+
+    Returns:
+        The prefix allowed function.
+    """
+    prefix_allowed_tokens_fns = list()
+
+    if dataset_config.task == NER and isinstance(tokenizer, PreTrainedTokenizer):
+        parser = get_ner_parser(dataset_config=dataset_config)
+        json_prefix_allowed_tokens_fn = build_transformers_prefix_allowed_tokens_fn(
+            tokenizer_data=tokenizer, character_level_parser=parser
+        )
+
+        forbidden_token_ids = list()
+        forbidden_tokens = ["\n", "\n\n", "\n\n\n", "\t", "\t\t", "\t\t\t"]
```

Comment:
I am worried that for some tokenizers, this might not cover all the tokens that express new lines etc. See the discussion here on the llama tokenizers and their context dependence. I encountered this when using the completiononly collator.

https://huggingface.co/docs/trl/sft_trainer#using-tokenids-directly-for-responsetemplate